lib/Net/Async/Redis.pm

```diff
@@ -1096,6 +1096,48 @@ sub execute_command {
      ->retain;
 }
 
+=head2 RESP3 and RESP2 compatibility
+
+In RESP3 some of the responses are structured differently from RESP2 
+L<Net::Async::Redis> guarantees the same structure unless you have explicitly requested the new types
+
+=cut
+
+around [qw(xread xreadgroup)] => async sub {
+    my ($code, $self, @args) = @_;
+    return await $self->$code(@args) if $self->{hashrefs};
+
+    my $response = await $self->$code(@args);
+
+    # protocol_level is detected while connecting checking before this point is wrong.
+    return $response if $self->{protocol_level} eq 'resp2';
```

Comment:
I think this can be changed, maybe we can have one return for both of conditions
```suggestion
    my $response = await $self->$code(@args);

    # protocol_level is detected while connecting checking before this point is wrong.
    return $response if $self->{protocol_level} eq 'resp2' || $self->{hashrefs};
```